docs/topics/settings.rst

```diff
@@ -234,6 +234,55 @@ example, proper setting names for a fictional robots.txt extension would be
 ``ROBOTSTXT_ENABLED``, ``ROBOTSTXT_OBEY``, ``ROBOTSTXT_CACHEDIR``, etc.
 
 
+.. _component-lists:
+
+Component lists
+===============
+
+A **component list** is a :class:`dict` where keys are :ref:`components
```

Comment:
As you probably expected, this is the piece I'm least certain about :) I found it confusing to call a priority dictionary "list", both in docs and in the programmatic API. Not sure exactly what's a better name though; it seems other names are going to be longer ("component priority dictionary"), but it could be fine.

Reply:
Yes, this was the hardest part. I think I went with component list because of the existence of `build_component_list`, but what that does is take this as input and return an actual list, so if anything it seems to support the need for a different name.